utils/utils.go

```diff
@@ -1058,26 +1057,36 @@ func ForkVersionAtEpoch(epoch uint64) *types.ForkVersion {
 	}
 }
 
-func LogError(infoIdentifier string, err error) *logrus.Entry {
-	var fileIdentifier, lineIdentifier, functionIdentifier string
+func LogFatal(err error, errorMsg interface{}, callerSkip int, infoIdentifiers ...string) {
+	logErrorInfo(err, callerSkip, infoIdentifiers...).Fatal(errorMsg)
+}
+
+func LogError(err error, errorMsg interface{}, callerSkip int, infoIdentifiers ...string) {
+	logErrorInfo(err, callerSkip, infoIdentifiers...).Error(errorMsg)
+}
 
-	pc, fullFilePath, line, ok := runtime.Caller(1)
+func logErrorInfo(err error, callerSkip int, infoIdentifiers ...string) *logrus.Entry {
+	logFields := logrus.NewEntry(logrus.New())
+
+	pc, fullFilePath, line, ok := runtime.Caller(callerSkip + 2)
 	if ok {
-		fileIdentifier = filepath.Base(fullFilePath)
-		functionIdentifier = runtime.FuncForPC(pc).Name()
-		lineIdentifier = strconv.Itoa(line)
+		logFields = logFields.WithFields(logrus.Fields{
+			"file":     filepath.Base(fullFilePath),
+			"line":     line,
+			"function": runtime.FuncForPC(pc).Name(),
+		})
 	} else {
-		msg := "Cannot read callstack"
-		fileIdentifier = msg
-		functionIdentifier = msg
-		lineIdentifier = msg
+		logFields = logFields.WithField("runtime", "Callstack cannot be read")
+	}
+
+	if err != nil {
+		logFields = logFields.WithField("error type", fmt.Sprintf("%T", err)).WithError(err)
+	}
+
+	infoCount := 0
+	for _, info := range infoIdentifiers {
+		logFields = logFields.WithField(fmt.Sprintf("info_%v", infoCount), info)
 	}
```

Comment:
`infoCount` is not increment, so passing more than 1 `infoIdentifiers` will only show the last one labeled `info_0`.

```suggestion
	for i, info := range infoIdentifiers {
		logFields = logFields.WithField(fmt.Sprintf("info_%v", i), info)
	}
```

Reply:
Ah, sorry for that. Fixed.